2016/proposals/deng-pan.md

```diff
@@ -0,0 +1,130 @@
+#Integrating `pandas.Panel` and xarray Features
+
+## Abstract
+
+Pandas package excels at processing tabular models, especially 1-D and 2-D models with data structure `pd.Series` and `pd.DataFrame`. However, despite the specialized data structure for 3-D and N-D data, `pd.Panel`, pandas is not well-designed for high dimensional data processing. Xarray was then developed to compensate the weak part of pandas. Xarray package enabled statistic and mathematical analysis of high-dimensional data, and implemented several new features that are missing in pandas.
+ 
+However, some useful features in pandas were not implemented in xarray yet. Thus, my project will be focusing on porting features from pandas to xarray, and improving data structure conversion between pandas and xarray. By the end of my work, I expect pandas/xarray users can have specialized tools for different types of data processing, and migrate between pandas and xarray smoothly. 
+
+## Technical Details
+
+Most of my proposal is supposed to be carried out with current implemented features in pandas and xarray. For PCA part, to improve the performance, I might switch to C++ and Eigen3 library.
```

Comment:
currently ``xarray`` is pure python, this would be a very large change. using ``cython`` and/or ``numba`` *might* be acceptable. I suspect that ``xarray`` will eventually have to go down this route. @shoyer can shed some more light.